src/platform/silabs/multi-ota/OTAHooks.cpp

```diff
@@ -16,39 +16,55 @@
  *    limitations under the License.
  */
 
-#include <include/platform/CHIPDeviceLayer.h>
 #include <platform/silabs/multi-ota/OTAMultiImageProcessorImpl.h>
 
 #include <app/clusters/ota-requestor/OTARequestorInterface.h>
 
 #include <platform/silabs/multi-ota/OTAFactoryDataProcessor.h>
+
+#ifndef SLI_SI91X_MCU_INTERFACE
 #include <platform/silabs/multi-ota/OTAFirmwareProcessor.h>
+#endif
+
+#if (SLI_SI91X_MCU_INTERFACE | EXP_BOARD)
+#include <platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.h>
+#endif
 
 #if OTA_TEST_CUSTOM_TLVS
 #include <platform/silabs/multi-ota/OTACustomProcessor.h>
 #endif
 
 CHIP_ERROR chip::OTAMultiImageProcessorImpl::ProcessDescriptor(void * descriptor)
 {
-    [[maybe_unused]] auto desc = static_cast<chip::OTAFirmwareProcessor::Descriptor *>(descriptor);
+    [[maybe_unused]] auto desc = reinterpret_cast<chip::OTATlvProcessor::Descriptor *>(descriptor);
     ChipLogDetail(SoftwareUpdate, "Descriptor: %ld, %s, %s", desc->version, desc->versionString, desc->buildDate);
 
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR chip::OTAMultiImageProcessorImpl::OtaHookInit()
 {
-    static chip::OTAFirmwareProcessor sApplicationProcessor;
     static chip::OTAFactoryDataProcessor sFactoryDataProcessor;
 
-    sApplicationProcessor.RegisterDescriptorCallback(ProcessDescriptor);
+    auto & imageProcessor = chip::OTAMultiImageProcessorImpl::GetDefaultInstance();
     sFactoryDataProcessor.RegisterDescriptorCallback(ProcessDescriptor);
+    ReturnErrorOnFailure(imageProcessor.RegisterProcessor(OTAProcessorTag::kFactoryDataProcessor, &sFactoryDataProcessor));
 
-    auto & imageProcessor = chip::OTAMultiImageProcessorImpl::GetDefaultInstance();
-    ReturnErrorOnFailure(
-        imageProcessor.RegisterProcessor(static_cast<uint32_t>(OTAProcessorTag::kApplicationProcessor), &sApplicationProcessor));
-    ReturnErrorOnFailure(
-        imageProcessor.RegisterProcessor(static_cast<uint32_t>(OTAProcessorTag::kFactoryDataProcessor), &sFactoryDataProcessor));
+#ifdef SLI_SI91X_MCU_INTERFACE
+    // 917SOC: TA is the OTA processor for application image
+    static chip::OTAWiFiFirmwareProcessor sApplicationProcessor;
+#else  // EFR32
+    static chip::OTAFirmwareProcessor sApplicationProcessor;
+#endif // SLI_SI91X_MCU_INTERFACE
```

Comment:
This feels like a template class <T> could benefit the implementation. @mkardous-silabs @lpbeliveau-silabs  what do you think? 

Can be a follow-up task

Reply:
I have created for follow up for miscellaneous items pending on multi-ota feature. Added this also in follow up task - https://jira.silabs.com/browse/MATTER-4992.